tests/test_math/test_parameter_set.py

```diff
@@ -84,3 +84,25 @@ def test_to_string(self):
         assert ps.to_string(1) == "1.2, 2.3, 3.5"
         assert ps.to_string(3) == "1.234, 2.346, 3.457"
         assert ps.to_string(10) == "1.2345, 2.3456, 3.4567"
+
+    def test_get_item(self):
+        const1 = Constant(1, "c1")
+        const2 = Constant([2, 3, 4], "c2")
+        var1 = Variable(5, "v1")
+        var2 = Variable([6, 7, 8], "v2")
+
+        ps = ParameterSet()
+        ps.add_parameter(const1)
+        ps.add_parameter(const2)
+        ps.add_parameter(var1)
+        ps.add_parameter(var2)
+
+        assert np.allclose(ps[0].constants["c1"].value, 1)
+        assert np.allclose(ps[0].constants["c2"].value, 2)
+        assert np.allclose(ps[0].variables["v1"].value, 5)
+        assert np.allclose(ps[0].variables["v2"].value, 6)
+
+        assert np.allclose(ps[1, 2].constants["c1"].value, 1)
+        assert np.allclose(ps[1, 2].constants["c2"].value, [3, 4])
+        assert np.allclose(ps[1, 2].variables["v1"].value, 5)
+        assert np.allclose(ps[1, 2].variables["v2"].value, [7, 8])
```

Comment:
I added two asserts that should check if parameters are actually removed.

```suggestion
        assert np.allclose(ps[1, 2].variables["v1"].value, 5)
        assert np.allclose(ps[1, 2].variables["v2"].value, [7, 8])
        
        assert len(ps[5, 6].constants) == 0
        assert "v1" not in ps[6,7].variables
```